File: solutions/projects/proj10/proj10_start/Apple.java
public class Apple {
    private int count;

    public Apple() {
        count = 1;
    }

    public String getName() {
        return "Apple";
    }

    public void setCount(int num) {
        //only allow counts between 1-10
        if (num >= 1 && num <= 10) {
            count = num;
        }
    }

    public int getCount() {
        return count;
    }

    public double getUnitPrice() {
        return 0.85;
    }

    public double getFullPrice() {
        return count*getUnitPrice();
    }

    public String toString() {
        return String.format("%s, count: %d, price: $%.2f", getName(), getCount(), getFullPrice());
    }
}

File: solutions/projects/proj11/Apple.cs
public class Apple : CountableItem {

    public Apple() {
        Count = 1;
    }

    public String Name => "Apple";

    private int _count = 1;
    public int Count {
        get => _count;
        set {
            if (value >= 1 && value <= 10) {
                _count = value;
            }
        }
    }

    public double UnitPrice => 0.85;

    public double FullPrice => Count*UnitPrice;

    public override string ToString() {
        return $"{Name}, count: {Count}, price: {FullPrice:C}";
    }
}